controllers/extensionRequests.js

```diff
@@ -14,73 +14,156 @@ const { parseQueryParams } = require("../utils/queryParser");
+      return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
     }
+  } else {
+    try {
+      const extensionBody = req.body;
 
-    const extensionRequest = await extensionRequestsQuery.createExtensionRequest(extensionBody);
+      let assigneeUsername = await getUsernameElseUndefined(extensionBody.assignee);
+      let assigneeId = extensionBody.assignee;
+      if (!assigneeUsername) {
+        assigneeId = await getUserIdElseUndefined(extensionBody.assignee);
+        assigneeUsername = extensionBody.assignee;
+        extensionBody.assignee = assigneeId;
+      }
 
-    const extensionLog = {
-      type: "extensionRequests",
-      meta: {
+      if (!assigneeId) {
+        return res.boom.badRequest("User with this id or username doesn't exist.");
+      }
+
+      if (req.userData.id !== extensionBody.assignee && !req.userData.roles?.super_user) {
+        return res.boom.forbidden("Only assigned user and super user can create an extension request for this task.");
+      }
+
+      const { taskData: task } = await tasks.fetchTask(extensionBody.taskId);
+      if (!task) {
+        return res.boom.badRequest("Task with this id or taskid doesn't exist.");
+      }
+      if (task.assignee !== assigneeUsername) {
+        return res.boom.badRequest("This task is assigned to some different user");
+      }
+      if (task.endsOn >= extensionBody.newEndsOn) {
+        return res.boom.badRequest("The value for newEndsOn should be greater than the previous ETA");
+      }
+      if (extensionBody.oldEndsOn !== task.endsOn) {
+        extensionBody.oldEndsOn = task.endsOn;
+      }
+
+      const prevExtensionRequest = await extensionRequestsQuery.fetchExtensionRequests({
         taskId: extensionBody.taskId,
-        createdBy: req.userData.id,
-      },
-      body: {
-        extensionRequestId: extensionRequest.id,
-        oldEndsOn: task.endsOn,
-        newEndsOn: extensionBody.newEndsOn,
         assignee: extensionBody.assignee,
-        status: EXTENSION_REQUEST_STATUS.PENDING,
-      },
-    };
+      });
+      if (prevExtensionRequest.length) {
+        return res.boom.forbidden("An extension request for this task already exists.");
+      }
 
-    await addLog(extensionLog.type, extensionLog.meta, extensionLog.body);
+      const extensionRequest = await extensionRequestsQuery.createExtensionRequest(extensionBody);
 
-    return res.json({
-      message: "Extension Request created successfully!",
-      extensionRequest: { ...extensionBody, id: extensionRequest.id },
-    });
-  } catch (err) {
-    logger.error(`Error while creating new extension request: ${err}`);
-    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+      const extensionLog = {
+        type: "extensionRequests",
+        meta: {
+          taskId: extensionBody.taskId,
+          createdBy: req.userData.id,
+        },
+        body: {
+          extensionRequestId: extensionRequest.id,
```

Comment:
extensionBody object may not contain the id property yet, since it is only assigned after the extensionRequest object is created. This means that if the extensionBody object does not contain the id property, the ...extensionBody spread operator will copy all of the properties from the extensionBody object to the extensionRequest object, except for the id property. This will result in the extensionRequest object being missing the id property.

Reply:
Since the await is used it would wait for the extensionRequest to create hence it would contain the id property. So null wont be there and  extension request will always contain the id field and all other details.